pkg/controller/windowsmachineconfig/windows/windows.go

```diff
@@ -90,6 +103,94 @@ func (vm *Windows) initializeBootstrapperFiles() error {
 	return nil
 }
 
+// handleHybridOverlay ensures that the hybrid overlay is running on the node
+func (vm *Windows) handleHybridOverlay(nodeName string) error {
+	// Check if the hybrid-overlay is running
+	_, stderr, err := vm.Run("Get-Process -Name \""+hybridOverlayProcess+"\"", true)
+
+	// stderr being empty implies that hybrid-overlay was running.
+	if err == nil || stderr == "" {
+		// Stop the hybrid-overlay
+		stopCmd := "Stop-Process -Name \"" + hybridOverlayProcess + "\""
+		_, stderr, err := vm.Run(stopCmd, true)
+		if err != nil || stderr != "" {
+			log.Info("%s:\n%s", stopCmd, stderr)
+			return errors.Wrap(err, "unable to stop hybrid-overlay")
+		}
+	}
+
+	_, stderr, err = vm.Run(mkdirCmd(logDir), false)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("unable to create %s directory:\n%s", logDir, stderr))
+	}
+
+	if err := vm.CopyFile(wkl.HybridOverlayPath, remoteDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error copying %s-->%s", wkl.HybridOverlayPath,
+			remoteDir+wkl.HybridOverlayName))
+	}
+
+	// Start the hybrid-overlay in the background over ssh. We cannot use vm.Run() and by extension WinRM.Run() here as
+	// we observed WinRM.Run() returning before the commands completes execution. The reason for that is unclear and
+	// requires further investigation.
+	go vm.RunOverSSH(remoteDir+wkl.HybridOverlayName+" --node "+nodeName+
```

Comment:
nit: fyi while running locally last week I noticed that we may be leaking resources because of go routines since we're running within a controller. We may have to use [waitGroup](https://blog.golang.org/pipelines) or something.

Reply:
> As in you pulled this code and were running with it locally? 

Yes, I think it is because of vm.RunOverSSH can be run for each request that is coming in but it's not something really pressing atm as the memory usage increase with the number of Windows VMs we're managing.

Reply:
I have added a TODO with a Jira link